### PR TITLE
refactor(provider/deepgram): migrate transcription tests to fixture pattern

### PR DESCRIPTION
**Pull request by dancer**
## background

part of the ongoing effort to normalize provider tests to use real API response fixtures instead of inline mock data (#12270)

## summary

- record real transcription fixture from deepgram nova-3 API with language detection enabled
- replace inline `prepareJsonResponse` helper with `prepareJsonFixtureResponse` that reads from `__fixtures__/deepgram-transcription.json`
- use `beforeEach` with fixture for standard tests, inline `server.urls` for edge cases (custom detected language, undefined language)
- speech model test unchanged (uses `prepareAudioResponse` for binary audio data, not part of fixture migration scope)

## verification

<details>
<summary>tests pass</summary>

```
 ✓ src/deepgram-error.test.ts  (1 test) 2ms
 ✓ src/deepgram-transcription-model.test.ts  (9 tests) 31ms
 ✓ src/deepgram-speech-model.test.ts  (15 tests) 35ms

 Test Files  3 passed (3)
      Tests  25 passed (25)
```
</details>

<details>
<summary>no old helpers remain</summary>

```
$ grep -rc 'prepareJsonResponse\|prepareStreamResponse' packages/deepgram/src/
(all zero)
```
</details>

## checklist

- [x] tests have been added / updated (for bug fixes / features)
- [ ] documentation has been added / updated (for bug fixes / features)
- [ ] a _patch_ changeset for relevant packages has been added (run `pnpm changeset` in root)
- [x] i have reviewed this pull request (self-review)

## related issues

ref #12270

### CODE REVIEW

File: packages/deepgram/src/deepgram-transcription-model.test.ts [hunk 1]
```diff
@@ -17,233 +18,209 @@ const server = createTestServer({
...
+          currentDate: () => testDate,
+        },
+      });
+
+      const result = await customModel.doGenerate({
+        audio: audioData,
+        mediaType: 'audio/wav',
+      });
+
+      expect(result.response).toMatchObject({
```
Comment (gr2m) [inline packages/deepgram/src/deepgram-transcription-model.test.ts]: should we use `.toMatchSnapshot()` for responses since fixtures are generated? Will make it easier to update when fixtures change

**Review by gr2m — APPROVED** [review]
